src/main/java/uk/co/ramp/covid/simulation/population/Population.java

```diff
@@ -247,17 +247,20 @@ public void allocatePeople() {
 
     // This method assigns a random number of neighbours to each Household
     public void assignNeighbours() {
-        for (int i = 0; i < this.nHousehold; i++) {
-            Household cHouse = this.population[i];
+        for (Household cHouse : population) {
             int expectedNeighbours = PopulationParameters.get().getExpectedNeighbours();
             int nneighbours = (int) rng.nextPoisson(expectedNeighbours);
-            int[] neighbourArray = new int[nneighbours];
             for (int k = 0; k < nneighbours; k++) {
-                int nInt = rng.nextInt(0, this.nHousehold - 1);
-                if (nInt == i) k--;
-                else neighbourArray[k] = nInt;
+
+                Household neighbour = population[rng.nextInt(0, population.length - 1)];
+
+                if (neighbour == cHouse) {
+                    k--;
+                    continue;
+                }
+
+                cHouse.addNeighbour(neighbour);
```

Comment:
It looks possible we could end up with duplicates in the list of neighbours.

Reply:
Another good spot, suggest we move the List of neighbours to a Set to avoid this (and there's no ordering on neighbours anyway). I'll fix this up too